### PR TITLE
[docs] Add functions-deploy guide for Pulsar Functions (new)

### LINKED ISSUE #4554: [docs] Improve the structure of Pulsar Functions
**Is your feature request related to a problem? Please describe.**
I'm a little confused when searching information on Pulsar Functions. The content is scattered in different sections and does not integrate as a whole. Meanwhile, some information duplicates in *Overview* and other sections. It's a little bit frustrating to read the docs as a new user.

**Describe the solution you'd like**
I want to re-structure the architecture of Pulsar Functions as follows, so it will be easier to read and find information.
image

If you have any suggestion on the new architecture, feel free to contact me.
If there is no issue with it, I'll update the content throughout.

### PR DESCRIPTION
**Pull request by Jennifer88huang**
Fixes #4554 

### Motivation
Add a new deploy tutorial for new Pulsar Functions structure.

### Modifications
Based on the original functions-deploying file, re-adjust the structure and language of functions-deploy.

### CODE REVIEW

**Comment by Jennifer88huang** [comment]
@wolfstudy Could you also please help review the content? Thank you very much.

File: site2/docs/functions-deploy.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+## Requirements
+
+To deploy and manage Pulsar Functions, you need to have a Pulsar cluster running. There are several options for this:
+
+* You can run a [standalone cluster](getting-started-standalone.md) locally on your own machine.
+* You can deploy a Pulsar cluster on [Kubernetes](deploy-kubernetes.md), [Amazon Web Services](deploy-aws.md), [bare metal](deploy-bare-metal.md), [DC/OS](deploy-dcos.md), and more.
+
+If you run a non-[standalone](reference-terminology.md#standalone) cluster, you need to obtain the service URL for the cluster. How you obtain the service URL depends on how you deploy your Pulsar cluster.
+
+If you want to deploy and trigger Python user-defined functions, you need to install [the pulsar python client](http://pulsar.apache.org/docs/en/client-libraries-python/) first.
```
Comment (sijie) [inline site2/docs/functions-deploy.md]: you need to install [the pulsar python client](http://pulsar.apache.org/docs/en/client-libraries-python/) on all the machines running [functions workers](link-to-functions-worker).

File: site2/docs/functions-deploy.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+$ bin/pulsar-admin functions update \
+  --py myfunc.py \
+  --classname myfunc.SomeFunction \
+  --inputs persistent://public/default/new-input-topic \
+  --output persistent://public/default/new-output-topic
+```
+
+### Parallelism
+
+Pulsar Functions run as processes, which is called **instances**. When you run a Pulsar Function, it runs as a single instance by default. In [local run mode](#local-run-mode), you can *only* run a single instance of a function.
```
Comment (sijie) [inline site2/docs/functions-deploy.md]: It can be run as threads as well. so I think it can be changed to `Pulsar Functions run as processes or threads`

Comment (sijie) [reply site2/docs/functions-deploy.md]: > In [local run mode](#local-run-mode), you can *only* run a single instance of a function.

Each localrun command runs only a single instance of a function. If you want to run multiple instances, you can execute localrun multiple times.

Comment (Jennifer88huang) [reply site2/docs/functions-deploy.md]: both processes or threads are called instances?
or only processes are called instances?

**Comment by Jennifer88huang** [comment]
Updated as per advice from @sijie . PTAL again.

**Comment by Jennifer88huang** [comment]
If there is no other issue, could we merge this PR before the end of July?
If there is issue, feel free to comment. Thank you.

**Comment by sijie** [comment]
retest this please

**Comment by Jennifer88huang** [comment]
run integration tests
run java8 tests

**Comment by Jennifer88huang** [comment]
run integration tests
run java8 tests

**Comment by Jennifer88huang** [comment]
run java8 tests

**Comment by Jennifer88huang** [comment]
run java8 tests

**Pull request merged**